packages/telemetry/src/slog-to-otel.js

```diff
@@ -412,45 +479,61 @@ export const makeSlogToOtelKit = (tracer, overrideAttrs = {}) => {
         if (!kd) {
           break;
         }
-        const crankSpan = spans.get(getCrankKey());
-        crankSpan.setAttributes(cleanAttrs({ delivery, ...attrs }));
+
+        const name = `deliver-${kd[0]}`;
+        const activeDeliverySpan = spans.push(
+          [name, attrs.vatID, attrs.deliveryNum],
+          { attributes: {} },
+        );
+        activeDeliverySpan.setAttributes(cleanAttrs({ delivery, ...attrs }));
         if (kd[0] === 'notify') {
           // Track call graph.
           const [_type, [notification]] = kd;
           // TODO: We only track the first notified kernel promise. Perhaps we should do more?
           const [kpId, { state }] = notification;
           const error = state === 'rejected' ? 'rejected' : undefined;
-          if (kernelPromiseToSendingCause.has(kpId)) {
+          // TODO: figure out if this is correct?
+          if (!isReplaying && kernelPromiseToSendingCause.has(kpId)) {
             // Track this notification as the cause for the current crank.
             const cause = JSON.parse(kernelPromiseToSendingCause.get(kpId));
+            // crankNum doesn't exist on replay
             crankNumToCause.init(slogAttrs.crankNum, {
               ...cause,
               attrs: { ...cause.attrs, state },
               error,
             });
           }
         } else if (kd[0] === 'message') {
-          // This is where the message is delivered.
-          // Track call graph.
           const [
             _type,
             target,
             { methargs, method = extractMethod(methargs), result },
           ] = kd;
+          const details = `E(${target}).${method}`;
+          const msgAttrs = {
+            'msg.target': target,
+            'msg.method': extractMethod(methargs),
+            'msg.slots': methargs.slots.join(','),
+            'msg.result': result,
+            details,
+          };
+          activeDeliverySpan.setAttributes(
+            cleanAttrs({ ...attrs, ...msgAttrs }),
+          );
+          // This is where the message is delivered.
+          // Track call graph.
           let cause;
           if (kernelPromiseToSendingCause.has(result)) {
             cause = JSON.parse(kernelPromiseToSendingCause.get(result));
           } else {
             // Create a new root span.
-            const name = `E(${target}).${method}`;
+            const crankSpan = spans.get(getCrankKey());
```

Comment:
I ended up having to unify the crank span with the deliver span by starting and ending the deliver span from `crank-start` and `crank-finish`, and then having `deliver` and `deliver-result` add attributes to it.  I'm not sure this is ideal but I was at a loss to do it any other way since some of the `clist` events that need to be part of this span happen outside the bounds of `deliver` and `deliver-result`.  It's a little ugly but the output it produces is actually nice.  I think this is as far as I'm prepared to take it for now.